src/main/java/org/mskcc/cbio/portal/scripts/UpdateCaseListsSampleIds.java

```diff
@@ -0,0 +1,216 @@
+                throw new RuntimeException(caseListFile.getAbsolutePath() + ": The following sample ids present in the case list file, but not specified in the clinical sample file: " + String.join(", ", extraSampleIds));
+            }
+            result.put(caseList.getStableId(), new LinkedHashSet<>(caseList.getSampleIds()));
+        }
+        return result;
+    }
+
+    private void updateCaseLists(Map<String, Set<String>> caseListSampleIdToSampleIds) {
+        // TODO Do we really have to do this? Is there a better way?
+        DaoCancerStudy.reCacheAll();
+        try {
+            for (Map.Entry<String, Set<String>> caseListStableIdToSampleIds: caseListSampleIdToSampleIds.entrySet()) {
+                String caseListStableId = caseListStableIdToSampleIds.getKey();
+                Set<String> sampleIds = caseListStableIdToSampleIds.getValue();
+                SampleList sampleList = daoSampleList.getSampleListByStableId(caseListStableId);
+                if (sampleList == null) {
+                    throw new RuntimeException("No case list with " + caseListStableId + " stable id is found");
+                }
+                LinkedHashSet<String> newCaseListSampleIds = new LinkedHashSet<>(sampleIds);
+                newCaseListSampleIds.addAll(sampleList.getSampleList());
+                ArrayList<String> newSampleArrayList = new ArrayList<>(newCaseListSampleIds);
+                sampleList.setSampleList(newSampleArrayList);
+                //TODO no need to run expensive db update if sampleList hasn't effectively changed
+                daoSampleList.updateSampleListList(sampleList);
+            }
+            CancerStudy cancerStudy = DaoCancerStudy.getCancerStudyByStableId(this.cancerStudyStableId);
+            List<SampleList> sampleLists = daoSampleList.getAllSampleLists(cancerStudy.getInternalId());
+            List<SampleList> remainingLists = sampleLists.stream().filter(sl ->
+                    !caseListSampleIdToSampleIds.containsKey(sl.getStableId()) && sl.getSampleList().stream().anyMatch(this.allSampleIds::contains)
+            ).collect(Collectors.toList());
+            for (SampleList remainingList: remainingLists) {
+                ArrayList<String> newSampleList = new ArrayList<>(remainingList.getSampleList());
+                newSampleList.removeAll(this.allSampleIds);
+                remainingList.setSampleList(newSampleList);
+                //TODO for optimization purpose we could supply to the update method 2 set of samples: samples that have to be added and samples that have to be removed
+                daoSampleList.updateSampleListList(remainingList);
+            }
+        } catch (DaoException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private LinkedHashSet<String> readSampleIdsFromDataFile(File dataFile) {
+        LinkedHashSet<String> allSampleIds = new LinkedHashSet<>();
+        FileReader reader = null;
+        try {
```

Comment:
this can be replaced with a try with resources. 

https://www.baeldung.com/java-try-with-resources